rts/Lua/LuaSyncedRead.cpp

```diff
@@ -8515,8 +8520,119 @@ int LuaSyncedRead::GetUnitScriptNames(lua_State* L)
 
 	return 1;
 }
+/***
+ *
+ * @function Spring.TraceRayUnits
+ * @number posX
+ * @number posY
+ * @number posZ
+ * @number dirX
+ * @number dirY
+ * @number dirZ
+ * @number traceLength
+ * @treturn {{len, Id},...}
+ */
+
+int LuaSyncedRead::TraceRayUnits(lua_State* L) //returns the list of units that an raytrace has hit
+{
+	float3 pos((double)luaL_checknumber(L, 1), (double)luaL_checknumber(L, 2), (double)luaL_checknumber(L, 3));
+	float3 dir((double)luaL_checknumber(L, 4), (double)luaL_checknumber(L, 5), (double)luaL_checknumber(L, 6));
+	float traceLength = luaL_checknumber(L, 7);
+	QuadFieldQuery qfQuery;
+	quadField.GetQuadsOnRay(qfQuery, pos, dir, traceLength);
+	CollisionQuery cq;
+	int num = 0;
+	lua_newtable(L);
+	for (const int quadIdx : *qfQuery.quads) {
+		const CQuadField::Quad& quad = quadField.GetQuad(quadIdx);
+
+		for (CUnit* u : quad.units) {
```

Comment:
It would be good if the returned list was sorted in order of distance, from the closest.

Reply:
Can you not reverse it in lua?